File: .NET/BinToBloom Backend/Models/Entities/PickupRequest.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BinToBloom_Backend.Models.Entities
{
    [Table("PickupRequests")]
    public class PickupRequest
    {
        [Key]
        [Column("pickup_id")]
        public int PickupId { get; set; }

        [Required]
        [Column("user_id")]
        public int UserId { get; set; }

        [Column("collector_id")]
        public int? CollectorId { get; set; }

        [Required]
        [Column("waste_type")]
        [StringLength(20)]
        [RegularExpression("^(FOOD|E-WASTE|BIODEGRADABLE|NON_BIODEGRADABLE)$", ErrorMessage = "Invalid waste type")]
        public string WasteType { get; set; } = string.Empty;


        [Column("notes")]
        [StringLength(500)]
        public string? Notes { get; set; }

        [Column("pickup_frequency")]
        [StringLength(10)]
        public string? PickupFrequency { get; set; }

        [Column("latitude", TypeName = "decimal(10, 8)")]
        public decimal? Latitude { get; set; }


        [Column("longitude", TypeName = "decimal(11, 8)")]
        public decimal? Longitude { get; set; }

        [Required]
        [Column("scheduled_date")]
        public DateOnly ScheduledDate { get; set; }

        [Required]
        [Column("scheduled_time")]
        public TimeOnly ScheduledTime { get; set; }

        [Column("pickup_status")]
        [StringLength(20)]
        [RegularExpression("^(PENDING|ACCEPTED|COMPLETED|PAID)$", ErrorMessage = "Status must be PENDING, ACCEPTED, COMPLETED or PAID")]
        public string PickupStatus { get; set; } = "PENDING";

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        [ForeignKey("CollectorId")]
        public Collector? Collector { get; set; }

        public ICollection<WasteLog> WasteLogs { get; set; } = new List<WasteLog>();
        public ICollection<TrackingLog> TrackingLogs { get; set; } = new List<TrackingLog>();

        public Payment? Payment { get; set; }
    }
}

File: Springboot/bintobloom-backend/src/main/java/com/bintobloom/entity/PickupRequest.java
package com.bintobloom.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "pickup_requests")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PickupRequest {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pickup_id")
    private Long pickupId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "password", "pickupRequests"})
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "collector_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "password", "pickupRequests"})
    private User collector;
    
    @NotNull(message = "Waste type is required")
    @Enumerated(EnumType.STRING)
    @Column(name = "waste_type", nullable = false, length = 50)
    private WasteType wasteType;
    
    @NotNull(message = "Scheduled date is required")
    @Column(name = "scheduled_date", nullable = false)
    private LocalDate scheduledDate;
    
    @NotNull(message = "Scheduled time is required")
    @Column(name = "scheduled_time", nullable = false)
    private LocalTime scheduledTime;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "pickup_status", nullable = false, length = 30)
    private PickupStatus pickupStatus = PickupStatus.PENDING;
    
    @Size(max = 500, message = "Notes must not exceed 500 characters")
    @Column(length = 500)
    private String notes;
    
    @Column(name = "pickup_frequency", length = 20)
    private String pickupFrequency;

    @Column(precision = 10, scale = 8)
    private java.math.BigDecimal latitude;

    @Column(precision = 11, scale = 8)
    private java.math.BigDecimal longitude;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
    
    public enum WasteType {
        FOOD_WASTE, E_WASTE, PLASTIC, PAPER, METAL, GLASS, MIXED, BIODEGRADABLE, NON_BIODEGRADABLE,
        ORGANIC_WASTE, RECYCLABLE_WASTE, CHEMICAL_WASTE, HAZARDOUS_WASTE, CONSTRUCTION_WASTE, NON_RECYCLABLE_COMMERCIAL
    }
    
    public enum PickupStatus {
        PENDING, ASSIGNED, PAYMENT_PENDING, PAID, IN_PROGRESS, COMPLETED, CANCELLED
    }
}